torch_ort/experimental/debug.py

```diff
@@ -0,0 +1,73 @@
+import os
+from torch_ort import ORTModule
+from onnxruntime.training.ortmodule._logger import LogLevel
+
+
+def save_intermediate_onnx_graphs(model: ORTModule, prefix: str) -> bool:
+    '''Saves all intermediate ONNX graphs produced by `model` at the specified `prefix`
+
+    When `prefix` is either None or an empty string, ONNX graphs saving is disabled, otherwise
+    multiple ONNX graphs will be saved as specified by `prefix` with an ending ".onnx" extension.
+
+    Args:
+        model (ORTModule): ORTModule instance
+        prefix(str): Full path plus a file prefix to be prepended to all ONNX graphs.
+                     Prefix must have a directory and file prefix components (e.g. /foo/bar_).
+                     The directory must exist and be writable by the user.
+
+    Returns:
+        True if ONNX graph save is enabled or False otherwise
+
+    Raises:
+        NotADirectoryError: if :attr:`prefix`does not contain a directory component
+        NameError: if :attr:`prefix`does not contain a file prefix component
+        TypeError: if :attr:`model` is not a :class:`ORTModule` object
+
+    '''
+
+    if not isinstance(model, ORTModule):
+        raise TypeError(f'`model` must be a `ORTModule` object, but `{type(model)}` object was specified.')
+
+    assert prefix is None or isinstance(prefix, str), '`prefix` must be a non-empty string or `None`.'
```

Comment:
Can prefix be ```None```? I just ran a quick test:
```py
>>> import os
>>> prefix = None
>>> dir_name = os.path.dirname(prefix)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/anaconda/envs/bmeswani_py38/lib/python3.8/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

Did you mean to assert that the prefix is ```not None```?